presto-hive/src/main/java/io/prestosql/plugin/hive/BackgroundHiveSplitLoader.java

```diff
@@ -302,11 +313,25 @@ private void invokeNoMoreSplitsIfNecessary()
     private ListenableFuture<?> loadPartition(HivePartitionMetadata partition)
             throws IOException
     {
-        String partitionName = partition.getHivePartition().getPartitionId();
+        HivePartition hivePartition = partition.getHivePartition();
+        String partitionName = hivePartition.getPartitionId();
         Properties schema = getPartitionSchema(table, partition.getPartition());
         List<HivePartitionKey> partitionKeys = getPartitionKeys(table, partition.getPartition());
         TupleDomain<HiveColumnHandle> effectivePredicate = compactEffectivePredicate.transform(HiveColumnHandle.class::cast);
 
+        List<HiveColumnHandle> partitionColumns = getPartitionKeyColumnHandles(table, typeManager);
+        BooleanSupplier partitionMatchSupplier = () -> {
+            TupleDomain<ColumnHandle> dynamicFilter = dynamicFilterSupplier.get();
+            if (dynamicFilter.isNone()) {
+                return false;
+            }
+            return partitionMatches(partitionColumns, dynamicFilter, hivePartition);
+        };
+        if (!partitionMatchSupplier.getAsBoolean()) {
+            // Avoid listing files and creating splits from a partition if it has been pruned due to dynamic filters
+            return COMPLETED_FUTURE;
+        }
```

Comment:
`partitionMatchSupplier` appears unnecessary (see comment below). In that case, this can turn into:

```java
        TupleDomain<ColumnHandle> dynamicFilter = dynamicFilterSupplier.get();
        if (!partitionMatches(partitionColumns, dynamicFilter, hivePartition)) {
            return COMPLETED_FUTURE;
        }
```

BTW, `partitionMatches` should return `false` if  `dynamicFilter` is `NONE`, so no need to special case it here.

Reply:
Currently `partitionMatches` doesn't handle `NONE` case since this check is done in `HivePartitionManager#getPartitions` for the normal partition pruning code flow, that's why I added it here.